lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -100,15 +105,17 @@ def preprocess(self, client_json: JSON) -> CONTEXT:
             dataset = Dataset.query(md5=run_id)
         except DatasetNotFound:
             self.logger.debug(f"Dataset with Run ID {run_id!r} not found")
```

Comment:
Now that we are raising an exception here, should we drop the debug statement?  The same information is in the exception raised.